segurata/enginee/enginee/spiders/auth.py

```diff
@@ -8,38 +8,46 @@
 from scrapy.exceptions import CloseSpider
 from scrapy.http import Response
 from scrapy.linkextractors import LinkExtractor
-from scrapy.spiders import CrawlSpider, Rule
+from scrapy.spiders import Spider
 
 from enginee.login.exceptions import *
 from enginee.login.login import check_login, login_request
 from enginee.user_config.exceptions import *
 from enginee.user_config.parse import user_config_parse
 
-Rules = Tuple[Rule]
 
+class AuthSpider(Spider):
+    '''Spider that logs in a website and scrapes its content.
+    
+    Arguments:
 
-class AuthSpider(CrawlSpider):
+        config {str} -- path to the configuration file.
+    
+    The configuration files has all the necessary configurations
+    like the credentials and DOM selectors to scrape the items.
+    
+    '''
     name = 'auth'
     start_urls = []
-    rules: ClassVar[Rules] = ()
 
-    def __init__(self, *args, **kwargs):
-        path_to_user_config = kwargs.get('config')
-        self._init_user_config(path_to_user_config)
-        # TODO: Assert that the config has the required credentials
-        self._populate_rules()
+    def __init__(self, *args, config, **kwargs):
+        user_config = user_config_parse(config)
+        self.config = user_config
         super().__init__(*args, **kwargs)
 
     def start_requests(self):
-        login_url = self.config_dict['login_url']
-        credentials = self.config_dict['credentials']
-        check_login_selector = self.config_dict['check_login_css']
-        after_login_url = self.config_dict['after_login_url']
+        config = self.config
+
+        login_url = config.get('login_url')
+        assert login_url is not None, 'Please input login URL.'
+
+        start_urls = config.get('start_urls')
+        assert start_urls is not None, 'Please input URLs to scrape.'
+        self.start_urls = start_urls
 
         login_info = {
-            'credentials': credentials,
-            'check_selector': check_login_selector,
-            'after_login_url': after_login_url,
+            'credentials': config['credentials'],
+            'check_selector': config['check_login_css_selector'],
```

Comment:
There is no `after_login_url` field here and `login` expects one. I guess this is yet to be implemented(?)

Reply:
Do we actually need this field?